build/build.sh

```diff
@@ -180,8 +188,27 @@ if [[ $dbg == 1 ]]; then
   mkdir -p $debug_dir
   cd $debug_dir
   if [[ $nocmake == 0 ]]; then
-    echo "$CMAKE -DRDI_CCACHE=$ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ../../src"
-    time $CMAKE -DRDI_CCACHE=$ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ../../src
+	if [[ $extratoolchain == 0 ]]; then
+		echo "$CMAKE -DRDI_CCACHE=$ccache"
+		echo "	-DCMAKE_BUILD_TYPE=Debug"
+		echo "	-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+		echo "	../../src"
+		time $CMAKE -DRDI_CCACHE=$ccache \
+			-DCMAKE_BUILD_TYPE=Debug \
+			-DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+			../../src
+	else
+		echo "$CMAKE -DRDI_CCACHE=$ccache"
+		echo "	-DCMAKE_BUILD_TYPE=Debug"
+		echo "	-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+		echo "	-DCMAKE_TOOLCHAIN_FILE=$toolchain"
+		echo "	../../src"
+		time $CMAKE -DRDI_CCACHE=$ccache \
+			-DCMAKE_BUILD_TYPE=Debug \
+			-DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+			-DCMAKE_TOOLCHAIN_FILE=$toolchain \
+			../../src
+	fi
```

Comment:
I believe you can simply always pass -DCMAKE_TOOLCHAIN_FILE=$toolchain without checking if the variable was specified.   If it is empty -DCMAKE_TOOLCHAIN_FILE will simply be ignored.  Please check.

I'd prefer to have this be one line commands as before.

Reply:
@stsoe The purpose of using multiple lines is to make the code rendering on github as well as vim editors much better - and if you have a very long single line, then even small changes would be very difficult to track.

Reply:
@yunqu  I hear you, but in this case I thought it made it less readable.  Thanks for fixing, it looks good now (at least to me) :-)